diff --git a/purl.install b/purl.install
index 9ee7e69..63a0714 100644
--- a/purl.install
+++ b/purl.install
@@ -1,58 +1,66 @@
 <?php
 
-function purl_schema()
-{
-    $schema = array();
-
-    $schema['purl_providers_settings'] = array(
-        'fields' => array(
-            'provider' => array(
-                'type' => 'varchar',
-                'length' => 255,
-                'not null' => true,
-            ),
-            'method' => array(
-                'type' => 'varchar',
-                'length' => 255,
-                'not null' => true,
-            ),
-            'settings' => array(
-                'type' => 'text',
-                'not null' => true,
-                'mysql_type' => 'mediumtext',
-            ),
-            'rebuild' => array(
-                'type' => 'int',
-                //'size' => 'tinyint',
-                'default' => 0,
-            ),
-        ),
-        'primary key' => array('provider'),
-        'description' => 'PURL provider settings',
+/**
+ * Implements hook_schema().
+ */
+function purl_schema() {
+  $schema['purl_providers_settings'] = array(
+    'fields' => array(
+      'provider' => array(
+        'description' => '',
+        'type' => 'varchar',
+        'length' => 255,
+        'not null' => true,
+      ),
+      'method' => array(
+        'description' => '',
+        'type' => 'varchar',
+        'length' => 255,
+        'not null' => true,
+      ),
+      'settings' => array(
+        'description' => '',
+        'type' => 'text',
+        'not null' => true,
+        'size' => 'medium',
+      ),
+      'rebuild' => array(
+        'description' => '',
+        'type' => 'int',
+        'size' => 'tiny',
+        'not null' => TRUE,
+        'default' => 0,
+      ),
+    ),
+    'primary key' => array(array('provider', 191)),
+    'description' => 'PURL provider settings',
     );
 
-    $schema['purl_modifiers'] = array(
-        'fields' => array(
-            'provider' => array(
-                'type' => 'varchar',
-                'length' => 255,
-                'not null' => true,
-            ),
-            'modifier' => array(
-                'type' => 'varchar',
-                'length' => 255,
-                'not null' => true,
-            ),
-            'value' => array(
-                'type' => 'text',
-                'not null' => true,
-                'mysql_type' => 'mediumtext',
-            ),
-        ),
-        'primary key' => array('provider', 'modifier'),
-        'description' => 'Index of PURL modifiers',
-    );
+  $schema['purl_modifiers'] = array(
+    'fields' => array(
+      'provider' => array(
+        'description' => '',
+        'type' => 'varchar',
+        'length' => 255,
+        'not null' => true,
+      ),
+      'modifier' => array(
+        'description' => '',
+        'type' => 'varchar',
+        'length' => 255,
+        'not null' => true,
+      ),
+      'value' => array(
+        'description' => '',
+        'type' => 'text',
+        'not null' => true,
+        'mysql_type' => 'mediumtext',
+      ),
+    ),
+    'primary key' => array(array('provider', 191), array('modifier', 191)),
+    'description' => 'Index of PURL modifiers',
+  );
 
-    return $schema;
+  return $schema;
 }
 
